### DISCUSSION TITLE
[4주차 알고리즘 스터디 버퍼1] 구현

### DISCUSSION THREAD

**Discussion by bigsaigon333** [fe-backpackers/js-algorithm-basecamp#50] [General]
## 알고리즘 스터디 회고
- 주차: 4주차
- 주제: 버퍼1
- 일시: 2021-10-24 Sun 22:00 ~ 23:00

## 회고 순서
1. 체크인 : ~22:05

**Comment by bigsaigon333** [comment]
## 체크인 

지금의 기분을 10점 만점으로 표현해주세요 :smile:

**Reply by sooster910** [reply]
7점 : 인구이동 어렵네요. 😭

**Reply by phoenix9373** [reply]
8점: 내일이 월요일이라는 것만 빼면 다 좋아요!

**Reply by bigsaigon333** [reply]
9점! 오랜만에 알고리즘 스터디 너무 좋습니다  버퍼 주때 하려고 했떤거 다 못해서 -1점 했습니다

**Reply by ganeodolu** [reply]
8점 : 주말을 알차게 보낸 것 같아서 좋네요 ㅎㅎ

**Reply by Yeonlisa** [reply]
7점 : 기출문제푸냐고 힘들었어요 😭😭😭😭😭

**Comment by bigsaigon333** [comment]
## 회고(버퍼주 계획 실천하셨나요~?)

@동동 
- leetCode 1문제 풀기
- 구현 알고리즘 실전 문제 1개 풀기
- DFS/BFS 알고리즘 실전 문제 2개 풀기

@리밋 
-  못푼문제 풀어보기

@로디 
- 3주차 DFS/BFS 내용 정리
- 2주차 구현 내용 정리

@judy.k 
- DFS/BFS 문제 풀기

@Sunmon 
- 못 풀고 지나갔던 문제 풀기
- stack, queue 자료구조 만들기

@연리사 
- 기출문제를 풀어보쟈!! 

@sue 
- DFS/BFS 알고리즘 문제 노출시키기

@호준 
-  못 풀었던 기출문제와 이해안됐던 문제 복습, 리뷰

**Reply by bigsaigon333** [reply]
### 계획 실천현황
- [x] leetCode 1문제 풀기
- [ ] 구현 알고리즘 실전 문제 1개 풀기
- [ ] DFS/BFS 알고리즘 실전 문제 2개 풀기: 하나밖에 못풀었네요...

### 주절주절...
- leetCode는 그냥 어떤 구조인지 확인할겸 쉬운거 2개 풀었습니다. 인터뷰 관련 모음집 같은게 있어서,  leetCode는 인터뷰 대비용으로 써도 좋을 것 같다는 생각이 드네요.
- 구현 알고리즘은 왤케 풀기 싫은지 몰겠네요... 뭔가 풀고 나도 개운하지 않고, 겨우 해냈네.. 이런 생각이 많이 들어서 좀 꺼려지는 부분인거 같습니다. 배움보다는 고난이 긴...

### 왜 DFS/BFS 알고리즘을 하나밖에 풀지 않았는가?
- 우선순위를 높게 설정하지 않았다. 시간이 남으면 풀어야지 하는 안일한 생각..
- 버퍼주라서 더 마음이 해이해졌던 것도 사실...

### Action Plan
- 요일과 시간을 정해놓고 알고리즘 공부를 해야겠다

**Reply by phoenix9373** [reply]
@로디

1. 3주차 DFS/BFS 내용은 정리하고, 추가적으로 PART 03 문제를 풀고 싶었는데! 오히려 실전에서 많이 풀게 됐다. 잊지 않게 한 주에 한 문제씩 풀어보자!
2. 2주차 구현 문제는 내용 정리보다 실전 문제 풀이에 집중했다. 게임 개발 문제가 의외로 잘 안 풀려서 어려웠다..! 구현 부분도 의도치 않게 실전에서 더 풀게 됐다. 깨달은 점은 시간 복잡도와 어떤 자료구조를 쓰면 좋을지 고민하고 풀이할 것.

**Reply by ganeodolu** [reply]
호준 : 복습은 못했지만 기출은 4문제 정도 풀었습니다. 어려워질수록 시간을 너무 많이 사용하지 않도록 유의해야할 것 같습니다. 틈틈히 다른분 코드도 잘 참고해서 미리미리 리뷰를 다는 것이 목표입니다.

**Reply by Yeonlisa** [reply]
### 원래 계획
- 기출문제 풀기
### 실제 한 것
- 기출문제 풀기
### 주저리 주저리
기출문제는 풀 수 있는건 다 풀었다.
다만 단원 넘어갈수록 풀은 문제가 줄어들었다.ㅜㅜ
DFS/BFS 알고리즘 너무 어렵다.

**Reply by Sunmon** [reply]
### 원래 계획

못 풀고 지나갔던 문제 풀기
stack, queue 자료구조 만들기

### 실제 한 것

못 푼 문제 1개 풀었다
자료구조 만들기 시간없어서 패스
알고리즘 스터디 하면서 얻게된 팁들 정리 😄 

### 계획과 한 것이 다른 이유

시간이 없었다


### Action Plan

회사 프로젝트 개발기간이 이번주까지라 야근하느라 바빠 돌아오는 알고리즘 스터디엔 참여를 못 할 것 같습니다...

**Comment by sooster910** [comment]
#### 계획
- dfs/bfs 문제 노출 

### 현황
- 예상했던 문제수보다 풀지 못해서 아쉽지만 그래도 스터디 그룹으로 혼자 공부하던 것보다는 의무감이 더 생겨 한 문제라도 더 보게되는것 같다. 
- 다음주도 dfs/bfs leetcode문제를 더 풀어볼 생각이다.

**Comment by bigsaigon333** [comment]
## 체크아웃 

지금의 기분을 10점 만점으로 표현해주세요 :smile:

**Reply by phoenix9373** [reply]
7점: 월요일이 50분 남아서 1점 깎였지만, 여전히 좋아요! 다음 주 sort 2문제 풀이 기대됩니다 ㅎㅎ😁😁

**Reply by Sunmon** [reply]
6 야근하기 싫어요

**Reply by bigsaigon333** [reply]
:sob:  선몬님 다음주 화이팅입니다 :+1: :+1: :+1: :+1:

**Reply by sooster910** [reply]
8점:  다음주 새로운 방식 기대되네요

**Reply by bigsaigon333** [reply]
10점  스터디 운영에 관한 많은 의견 주셔서 좋았습니다 :smile:  함께 만들어가는 스터디 :+1:

**Reply by Yeonlisa** [reply]
7점: 문제풀이가 일상이된거같아요 🧑‍🎓

**Reply by ganeodolu** [reply]
8점 : 월요일은 너무 금방 오네요ㅎㅎ 다시 sort 시작 !!!
